File: macos/Sources/SigilAuth/Models/PairHandshake.swift
import Foundation

public struct PairInitResponse: Codable, Equatable, Sendable {
    public let serverId: String
    public let serverPublicKey: String
    public let serverNonce: String
    public let expiresAt: Date

    enum CodingKeys: String, CodingKey {
        case serverId = "server_id"
        case serverPublicKey = "server_public_key"
        case serverNonce = "server_nonce"
        case expiresAt = "expires_at"
    }
}

public struct PairCompleteRequest: Codable, Equatable, Sendable {
    public let serverNonce: String
    public let clientPublicKey: String
    public let deviceInfo: DeviceInfo

    public struct DeviceInfo: Codable, Equatable, Sendable {
        public let name: String
        public let platform: String
        public let osVersion: String

        enum CodingKeys: String, CodingKey {
            case name
            case platform
            case osVersion = "os_version"
        }
    }

    enum CodingKeys: String, CodingKey {
        case serverNonce = "server_nonce"
        case clientPublicKey = "client_public_key"
        case deviceInfo = "device_info"
    }

    public init(serverNonce: String, clientPublicKey: String, deviceInfo: DeviceInfo) {
        self.serverNonce = serverNonce
        self.clientPublicKey = clientPublicKey
        self.deviceInfo = deviceInfo
    }
}

public struct PairCompleteResponse: Codable, Equatable, Sendable {
    public let status: String
    public let serverPublicKey: String
    public let pairedAt: Date

    enum CodingKeys: String, CodingKey {
        case status
        case serverPublicKey = "server_public_key"
        case pairedAt = "paired_at"
    }
}

public struct TrustedServer: Codable, Equatable, Sendable {
    public let serverUrl: URL
    public let serverId: String
    public let serverPublicKey: String
    public let serverFingerprint: String
    public let pairedAt: Date

    public init(
        serverUrl: URL,
        serverId: String,
        serverPublicKey: String,
        serverFingerprint: String,
        pairedAt: Date
    ) {
        self.serverUrl = serverUrl
        self.serverId = serverId
        self.serverPublicKey = serverPublicKey
        self.serverFingerprint = serverFingerprint
        self.pairedAt = pairedAt
    }
}

File: windows/src/Sigil.Windows.Core/Protocol/PairHandshake.cs
using System.Text.Json.Serialization;

namespace Sigil.Windows.Core.Protocol;

/// <summary>
/// Response from GET /pair/init.
/// Server sends ephemeral nonce + public key for mutual verification.
/// </summary>
public sealed record PairInitResponse(
    [property: JsonPropertyName("server_id")] string ServerId,
    [property: JsonPropertyName("server_public_key")] string ServerPublicKey,
    [property: JsonPropertyName("server_nonce")] string ServerNonce,
    [property: JsonPropertyName("expires_at")] string ExpiresAt
);

/// <summary>
/// Request payload for POST /pair/complete.
/// Client sends server nonce + own public key + device info.
/// </summary>
public sealed record PairCompleteRequest(
    [property: JsonPropertyName("server_nonce")] string ServerNonce,
    [property: JsonPropertyName("client_public_key")] string ClientPublicKey,
    [property: JsonPropertyName("device_info")] DeviceInfo DeviceInfo
);

/// <summary>
/// Device information sent during pairing.
/// </summary>
public sealed record DeviceInfo(
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("platform")] string Platform,
    [property: JsonPropertyName("os_version")] string OsVersion
);

/// <summary>
/// Response from POST /pair/complete.
/// Server confirms pairing with public key + timestamp.
/// </summary>
public sealed record PairCompleteResponse(
    [property: JsonPropertyName("status")] string Status,
    [property: JsonPropertyName("server_public_key")] string ServerPublicKey,
    [property: JsonPropertyName("paired_at")] string PairedAt
);
